inference-engine/thirdparty/clDNN/src/primitive_inst.cpp

```diff
@@ -51,19 +52,80 @@ void primitive_inst::check_memory_to_set(const memory& mem, const layout& layout
             if (layout.format.is_image_2d())
                 CLDNN_ERROR_MESSAGE(_node.id(), "Attempt to set user-supplied input or output buffer instead of an image");
             break;
+        case shared_mem_type::shared_mem_usm:
+            break;
         default:
             CLDNN_ERROR_MESSAGE(_node.id(), "Attempt to set user-supplied input or output memory of unknown/invalid type");
             break;
         }
     }
 }
 
-void primitive_inst::set_output_memory(memory::ptr mem) {
+void primitive_inst::set_output_memory(memory::ptr mem_new, bool check) {
+    auto& eng = _network.get_engine();
+    // skip all the buzz if no action actually required
+    if (eng.is_the_same_buffer(*mem_new, *_output)) {
```

Comment:
can't it be simplified to `mem_new == _output`?

Reply:
There might be different memory objects having the same buffer handle inside